vo_gpu_next: parse Dolby Vision metadata for dynamic scene brightness

quietvoid opened the pull request:
The changes would allow `vo=gpu-next` to have per-frame/scene dynamic brightness metadata coming from the Dolby Vision RPU.

I've changed the implementation to have the dependency in `libplacebo` instead.
Blocked by https://code.videolan.org/videolan/libplacebo/-/merge_requests/371 for now

Dudemanguy commented:
I have no opinions on this actual change but:

>I might have to add build support to the waf setup.

If this isn't straightforward, feel free to just not bother. Well, I don't know how this works but I'm assuming that checking for a rust library isn't going to "just work" in waf. That probably would probably require some re-engineering which at this stage which I think is out of scope for a build system that's basically on life support.

quietvoid commented:
> If this isn't straightforward, feel free to just not bother. Well, I don't know how this works but I'm assuming that checking for a rust library isn't going to "just work" in waf. That probably would probably require some re-engineering which at this stage I think is out of scope for a build system that's basically on life support.

It's C compatible so using pkgconfig should be straightforward.

kasper93 commented on video/out/vo_gpu_next.c:
```diff
@@ -714,7 +718,36 @@ static bool map_frame(pl_gpu gpu, pl_tex *tex, const struct pl_source_frame *src
+
+        if (header)
+            dovi_rpu_free_header(header);
+        dovi_rpu_free(rpu_in);
```
Why this asymmetry? You check for null in libdovi, so you could skip check here. Or... check both. Functionally no difference, just looks odd.

kasper93 commented on meson.build:
```diff
@@ -942,6 +942,12 @@ if features['libplacebo']
     endif
 endif
 
+dovi = dependency('dovi', required: false)
```
I think generally there is option added for each feature. `required: get_option('dovi')` or something should suffice, and entry in `meson_options.txt`

kasper93 replied:
also, probably version check for `libdovi` would be useful?

quietvoid replied:
> also, probably version check for `libdovi` would be useful?

The library isn't really distributed anywhere, and interfaces used here have been there forever.
So I don't think a minimum is necessary.

quietvoid commented:
I've added waf build support and fixed the condition in `vo_gpu_next.c` that made CI fail.
I might just make the PR ready, shouldn't have any negative impact.

haasn reviewed (approved):
LGTM otherwise

quietvoid commented on video/out/vo_gpu_next.c:
```diff
@@ -714,7 +718,36 @@ static bool map_frame(pl_gpu gpu, pl_tex *tex, const struct pl_source_frame *src
     }
-#endif
+
+#if PL_API_VER >= 246 && HAVE_LIBDOVI
```
I can certainly try to move it into libplacebo.  
My last attempt broke FFmpeg linking but I'll try a single header.

quietvoid replied:
Oh wait you meant `video/out/placebo/utils.c`.
Still the idea of having it in libplacebo would make it useful for non-mpv users.

quietvoid commented:
It's a bit better to have the dependency in libplacebo instead, so I changed the PR.
It requires https://code.videolan.org/videolan/libplacebo/-/merge_requests/371.
